### PR TITLE
TST/REF: parametrize arithmetic tests, simplify parts of core.ops

### PR DESCRIPTION
**Pull request by jbrockmendel**
Parametrization is straightforward.
Fixing xfailed test in test_properties is straightforward

In core.ops: ATM we have a nested closure that wraps all of na_op in a try/except, but there is really only one line we want to catch.  This simplifies the closure and catches only the relevant line.

### CODE REVIEW

File: pandas/core/ops.py [hunk 1]
```diff
@@ -1077,7 +1077,7 @@ def fill_binop(left, right, fill_value):
     return left, right
 
 
-def mask_cmp_op(x, y, op, allowed_types):
+def mask_cmp_op(x, y, op):
```
Comment (jreback) [inline pandas/core/ops.py]: can you add types where possible (not asking for everything of course, but if convenient / obvious)

**Comment by jbrockmendel** [comment]
It looks like hypothesis is having trouble here, but only in some builds

**Comment by jbrockmendel** [comment]
Punter on hypothesis breakage

**Comment by jreback** [comment]
lgtm. can you merge master again to make sure ok (as just merged the other 0dim change).

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**